### DISCUSSION TITLE
support for gwiot 7941e RFID reader

### DISCUSSION THREAD

**Discussion by FraatGIT** [arendst/Tasmota#20328] [Ideas]
### PROBLEM DESCRIPTION
I would like to integrate the gwiot 7941e module and continue to use scanned RFID codes.
I used the search function. However, nothing found that helps me.

There is a very small and nice a cheep rfid reader available with intergrated antena avaiable on the marked. 

* The RFID reader is using UART
* There is a Arduino library avaiable on github
   https://github.com/gutierrezps/gwiot7941e?tab=readme-ov-file
* A arduino reference is avaiable here:
 https://www.arduino.cc/reference/en/libraries/gwiot-7941e/

It would be nice if Tasmota was able to support this device and read the RFID.

### REQUESTED INFORMATION

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): Wemos D1 Mini
- [x] Tasmota binary firmware version number used: 13.3.0
  - [x] Pre-compiled
  - [] Self-compiled
- [x] Flashing tools used: Tasmota  Web Flasher

### ADDITIONAL CONTEXT
The RFID reader is avaiable here:
https://de.aliexpress.com/item/1005004482038419.html?spm=a2g0o.detail.0.0.53911ee9WsKzUi&gps-id=pcDetailTopMoreOtherSeller&scm=1007.40050.362094.0&scm_id=1007.40050.362094.0&scm-url=1007.40050.362094.0&pvid=77ac48f9-3ae5-4d5a-bbf0-b3bedd77297f&_t=gps-id:pcDetailTopMoreOtherSeller,scm-url:1007.40050.362094.0,pvid:77ac48f9-3ae5-4d5a-bbf0-b3bedd77297f,tpp_buckets:668%232846%238116%232002&pdp_npi=4%40dis%21EUR%212.52%212.22%21%21%212.74%21%21%40211b813f17037406223618929e73f7%2112000036725730112%21rec%21DE%211637493361%21

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by kobepower** [comment]
hey i just bought the d4 i cant find any software to make this work.

**Comment by smcgann99** [comment]
Should probably work if set for Wiegand output ?

**Comment by kobepower** [comment]
I have usb

**Reply by smcgann99** [reply]
I'm afraid that Tasmota is not really going to help you then

**Comment by kobepower** [comment]
i taught usb was best. so should i get widgand output not hard to do, i can
also go uart.

**Comment by kobepower** [comment]
i also own two m5stack.
https://www.aliexpress.com/item/1005007671968290.html?spm=a2g0o.productlist.main.3.5c9c5d59g1YzCk&algo_pvid=017bd56c-fd63-49ca-b6b3-6cdcbebbf2de&algo_exp_id=017bd56c-fd63-49ca-b6b3-6cdcbebbf2de-1&pdp_npi=4%40dis%21CAD%2110.52%219.99%21%21%217.38%217.01%21%402103011117306580421222237eddaf%2112000041741924706%21sea%21CA%216050513662%21X&curPageLogUid=7QLkqDBvwahB&utparam-url=scene%3Asearch%7Cquery_from%3A
should i buy this and this would help me read the module no

**Reply by smcgann99** [reply]
That will not help connecting reader to Tasmota. What is your application (what are you trying to do) ?

Tasmota supports D0 /D1 connection to GPIO if you build it with the Wiegand option.
define USE_WIEGAND                              // Add support for 24/26/32/34 bit RFID Wiegand interface (D0/D1) (+1k7 code)

It can then display the serial number of the card, that is all. If you want more features you are better off looking at something like - https://github.com/esprfid/esp-rfid.

In any case better if you open a new discussion rather than going off topic here.

**Comment by kobepower** [comment]
im just trying to read and write. i was told my gwiot d4 was plug and play,
but that not the case.

**Comment by FraatGIT** [comment]
I only know the 7941e. Can you please Share the link to the gwiot d4?

Fabian
________________________________
Von: kobepower ***@***.***>
Gesendet: Tuesday, November 5, 2024 6:53:23 AM
An: arendst/Tasmota ***@***.***>
Cc: FraatGIT ***@***.***>; Author ***@***.***>
Betreff: Re: [arendst/Tasmota] support for gwiot 7941e RFID reader (Discussion #20328)


im just trying to read and write. i was told my gwiot d4 was plug and play,
but that not the case.

**Comment by kobepower** [comment]
https://www.aliexpress.com/item/1005006947872163.html?spm=a2g0o.order_list.order_list_main.234.5a731802SVd2bC
the seller has software but he wont reply back. he asked if i wanted w26
uart or usb, i said usb since i didn know now the use the other but after
reading i figured it out when i open notepad i get hex numbers. the reader
works but its coded .

**Comment by kobepower** [comment]
i can change it to wigand
https://www.aliexpress.com/item/1005004945570425.html?spm=a2g0o.detail.pcDetailTopMoreOtherSeller.3.5728FIF7FIF7JL&gps-id=pcDetailTopMoreOtherSeller&scm=1007.40196.366991.0&scm_id=1007.40196.366991.0&scm-url=1007.40196.366991.0&pvid=4c64e3a5-cd60-4843-ad01-a8eb34a47df1&_t=gps-id:pcDetailTopMoreOtherSeller,scm-url:1007.40196.366991.0,pvid:4c64e3a5-cd60-4843-ad01-a8eb34a47df1,tpp_buckets:668%232846%238113%231998&pdp_npi=4%40dis%21CAD%215.30%214.83%21%21%213.73%213.40%21%402103241117307908975005603ed49e%2112000031109073485%21rec%21CA%216050513662%21X&utparam-url=scene%3ApcDetailTopMoreOtherSeller%7Cquery_from%3A

would any of these help me or would i get the same outcome.

**Comment by kobepower** [comment]
Okay. So the reason why I'm doing this is that since Bitcoin is getting
big, there's no real way to process it so I want to see if I can make a
credit card that can store Bitcoin and use it as a payment method so you
could have Bitcoin on it and then there'll be. A Bitcoin reader, and then
you just tap it like how you're tapping a regular bank card. Because if
bitcoin's going the way it's going, we're going to need some sort of way of
transferring and using bitcoin

**Reply by smcgann99** [reply]
OK so that's well and truly out of scope for Tasmota then. You would need to look elsewhere I'm afraid.
Tip you need to do some research on reading and writing to smart cards encryption etc etc ... good luck with your project.

**Comment by kobepower** [comment]
Thank you .

**Comment by kobepower** [comment]
[image: 20241106_000222.jpg]
can you tell what connection this is, im confused is uart wg26, i think i
found the problem the usb wire he sent me the pc dont reconize. i think im
gonna try new wire what connection is this thank  you
